packages/components/popover/src/Popover.test.tsx

```diff
@@ -1,27 +1,318 @@
+/* eslint-disable max-nested-callbacks */
+/* eslint-disable max-lines-per-function */
 import { render, screen } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
+import { mockResizeObserver } from 'jsdom-testing-mocks'
+import { useState } from 'react'
 import { describe, expect, it, vi } from 'vitest'
 
 import { Popover } from './Popover'
 
+const getSparkComponent = (componentName: string): Element | null => {
+  return document.querySelector(`[data-spark-component="${componentName}"]`)
+}
+
+mockResizeObserver()
+
 describe('Popover', () => {
-  it('should render', () => {
-    render(<Popover>Hello World!</Popover>)
+  describe('uncontrolled mode (stateful)', () => {
+    describe('opening', () => {
+      it('should open popover on click', async () => {
+        const user = userEvent.setup()
+
+        // Given a basic popover (closed by default)
+        render(
+          <Popover>
+            <Popover.Trigger asChild>
+              <button type="button">Click me</button>
+            </Popover.Trigger>
+            <Popover.Anchor>
+              <button type="button">Popover anchor</button>
+            </Popover.Anchor>
+            <Popover.Content>
+              <Popover.Arrow />
+              Popover content
+            </Popover.Content>
+          </Popover>
+        )
+
+        // ...by default, only trigger elements are rendered
+        expect(getSparkComponent('popover-anchor')).toBeInTheDocument()
+        expect(getSparkComponent('popover-trigger')).toBeInTheDocument()
+
+        // ...and content elements are not rendered
+        expect(getSparkComponent('popover-arrow')).toBeNull()
+        expect(getSparkComponent('popover-content')).toBeNull()
+        expect(screen.queryByText('Popover content')).not.toBeInTheDocument()
+
+        // When the user click on its associated trigger element
+        await user.click(screen.getByText('Click me'))
```

Comment:
```suggestion
        await user.click(screen.getByRole('button', { name: 'Click me' }))
```

Reply:
Yes :) I use `getByRole` only when I want to make sure of the element type. But in that case the button is not part of the Popover component I don't need to check its role (because getByRole is slower and the cause of performance issue on Leboncoin)

Reply:
For me is more related with readability purposes in this case but I get your point. Do you have any source related to `getByRole` being slower ?